internal/indexer/balances.go

```diff
@@ -73,3 +73,25 @@ func UpdateBalances(ptr *[]storage.Account, trx *chain.Transaction) error {
 	}
 	return nil
 }
+
+func (bc *blockContext) CheckIndexedBalances() error {
+	balances := bc.storage.GetAccounts()
+
+	for _, balance := range balances {
+		if balance.Balance.Cmp(big.NewInt(0)) == -1 {
+			return fmt.Errorf("balance of %s is negative", balance.Address)
+		}
+		balanceFromChain, err := bc.client.GetBalanceFromBlock(balance.Address, bc.block.Number)
```

Comment:
This gets should be concurrent. So first you need to get all of them to some array and then compare this two arrays